src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpointManagerBase.cs

```diff
@@ -12,11 +12,13 @@ namespace Microsoft.Azure.SignalR
 {
     internal abstract class ServiceEndpointManagerBase : IServiceEndpointManager
     {
+        // Endpoints for negotiation
         private readonly ConcurrentDictionary<string, IReadOnlyList<HubServiceEndpoint>> _endpointsPerHub = new ConcurrentDictionary<string, IReadOnlyList<HubServiceEndpoint>>();
 
         private readonly ILogger _logger;
 
-        public ServiceEndpoint[] Endpoints { get; }
+        // Filtered valuable endpoints from ServiceOptions
```

Comment:
>valuable [](start = 20, length = 8)

what does `valuable` mean?

Reply:
Follow original comments about this. Suppose means filter out duplicate endpoints from all available configuration sources.
> // select the most valuable endpoint with the same endpoint address


---
In reply to: [377514575](https://github.com/Azure/azure-signalr/pull/801#discussion_r377514575) [](ancestors = 377514575)